deployment/docker/build-3rd.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+# Run build process in docker container.
+
+set -ex -o pipefail
+
+CID=
+TMPDIR=$(mktemp -d)
+function docker_rm() {
+    if [[ -z "$CID" ]]; then
+        return 0
+    fi
+    if [[ -n "$LEAVE_CONTAINER" ]]; then
+        if [[ "${LEAVE_CONTAINER}" != "0" ]]; then
+            echo "Skip to clean container: ${CID}"
+            return 0
+        fi
+    fi
+    docker rm -f "$CID"
+}
+
+trap "docker_rm; rm -rf ${TMPDIR}" EXIT
+
+function docker_cp() {
+  local cid=${2%:*}
+  if [[ -z $cid ]]; then
+    # container -> host
+    docker cp $1 $2
+  else
+    # host -> container. Docker 1.7 or earlier does not support.
+    docker cp $1 $2 || {
+      local path=${2#*:}
+      tar -cO $1 | docker exec -i "${cid}" bash -c "tar -xf - -C ${path}"
+    }
+  fi
+}
+
+img_tag="3rd-build."
+if [[ -n "$JENKINS_HOME" ]]; then
+  # openvnet-axsh/branch1/el7
+  img_tag=$(echo "${img_tag}${JOB_NAME}/${BUILD_OS}" | tr '/' '.')
+else
+  img_tag="${img_tag}openvnet.$(git rev-parse --abbrev-ref HEAD).${BUILD_OS}"
+fi
+
+docker build -t "${img_tag}" -f "./deployment/docker/${BUILD_OS}-3rd-build.Dockerfile" .
```

Comment:
The ``docker`` command seems to have program name duplication with another X11 project. The script is designed to run and tested only on CentOS6 and 7 with a set of pre-installed packages. It will be hard to maintain the script to run on any base system.

We need a discussion the base line what the build script can assume or has to be satisfied.